### ISSUE TITLE
The "Overview" button should be labeled "Index"

### ISSUE THREAD

**Issue by plujon** [gollum/gollum#1591]
To me, the word "overview" implies a textual description of a subject written by a human being.  When one clicks the "Overview" button, the result is a directory listing.  That is commonly referred to as an index.

**Comment by heavywatal** [comment]
Agreed. I think "Latest Changes" and "Page History" can be better too.

**Comment by bartkamphorst** [comment]
I'm open to alternative button labels if they help clarify the respective button functions, but I think 'Index' is pretty nondescript tbh. In the context of gollum, it could even be misinterpreted as displaying the "git index". What about "File view" or "File listing" or "Document view"?

**Comment by dometto** [comment]
> it could even be misinterpreted as displaying the "git index". What about "File view" or "File listing" or "Document view"?

That's a good point. "File" is a bit too specific in Gollum, though (as it could be understood to refer to Files as opposed to Pages). "Wiki", "Wiki Contents", "Wiki Listing"? Open to more suggestions!

> Agreed. I think "Latest Changes" and "Page History" can be better too.

Open to suggestions for these two, too!

**Comment by benjaminwil** [comment]
In #1671 I will be touching the navigation menu templates, so maybe there's an opportunity to make label updates as well.

In my PR, I separate the menu buttons into button groups: global menu buttons (Overview, Latest Changes) are grouped together, separated from the current page menu buttons (Edit, Rename, Upload, History, New).

My proposal for button name changes would be:

- "Overview" changes to "Page Listing"
- "Latest Changes" changes to "Recent Changes" (this is Wikipedia nomenclature).
- "Page History" changes to "History" and is grouped with the Edit and Rename buttons.

Please let me know if these changes are acceptable for now.

In a future PR I would like to propose further separating the global buttons from the current page buttons. Maybe we could do this in a way where the global buttons are de-emphasized, since they are "context menu icons" not "action menu items", in my opinion.

Here's an example of what I mean:

![Screenshot from 2021-02-14 09-51-20](https://user-images.githubusercontent.com/883581/107884763-9ef12980-6eab-11eb-8579-8b750ae11bc0.png)

**Comment by dometto** [comment]
> "Overview" changes to "Page Listing"
    "Latest Changes" changes to "Recent Changes" (this is Wikipedia nomenclature).
    "Page History" changes to "History" and is grouped with the Edit and Rename buttons.

I like these suggestions. @bartkamphorst?

**Comment by bartkamphorst** [comment]
> I like these suggestions. @bartkamphorst?

On board!

**Comment by benjaminwil** [comment]
I just looked more deeply into making these changes. I wonder if it is too disruptive to make changes to the "Overview" and "Latest Changes" labels at this time.

The reason i say this is because "Overview" relates to the `/overview` route, and it's named this way widely across the codebase. It's the same for "Latest Changes", though to a lesser degree.

I can think of two clear ways forward:

1. Transition the code as well as the button labels.
  - This seems like some amount of work.
  - It would also require changes to `gollum-lib`.
2. Start using an internationalization/localization library, like `i18n`.
  - This way, we can continue to use `overview` and `latest_changes` to reference the features across the `gollum` and `gollum-lib` codebases, while de-coupling the features from the presented language in the frontend.
  - This also allows us to more easily provide the frontend in multiple languages.

I think both 1. and 2. would be worth doing, but I'm unsure whether 1. has a good cost-benefit ratio.

**Comment by dometto** [comment]
Localization support is an outstanding issue anyway: https://github.com/gollum/gollum/issues/1130

I agree that would be the ideal solution. There's a third option, I think: transition the buttons but not the route names. That would be mildly confusing for people reading the code, but not prohibitively so in my estimation. But I also don't think this issue (the labels of the minibuttons) is so pressing that it can't wait until we have time to properly address localization -- so happy to go for option 2 at some later point!
